app.py

```diff
@@ -11,700 +11,304 @@
-        
-    if trigger_id == "submission_timer" and submission_indicator_val == "Submitting... please wait":
-
-        # Checks all current schedule values
-        s = set( val[0] for dic in built_sched['props']['derived_virtual_data'] for val in dic.values())
-        
-        if " " not in s: # Scheduled blocks are blank in final schedule
-            return no_update, no_update, False, no_update
-        else:
-            return "Done", no_update, True, "sched"
-        
-    return no_update, no_update, True, no_update
-
-   
-@app.callback(
-    Output("built-sched", "children", allow_duplicate=True),
-    Output("error_card_body", "children"),
-    Output("tabs", "active_tab"),
-    Input("submission_indicator", "value"),
-    State("shifts-per-employee-slider", "value"),
-    State("employees-per-shift-slider", "value"),
-    State("checklist-input", "value"),
-    State("cons shifts", "value"),  # consecutive shifts allowed
-    State("initial-sched", "children"),
-    State("built-sched", "children"),
-    State("error_card_body", "children"),
-    prevent_initial_call=True
+def disp_initial_sched(
+    num_employees: int, rand_seed: int
+) -> tuple[DataFrame, DataFrame, bool, str]:
+    """Display initial availability schedule.
+
+    Display initial schedule in, and switch to, the availability
+    tab if number of employees or seed is changed.
+    """
+    # one less to account for trainee
+    num_employees -= 1
+
+    shifts = [str(i + 1) for i in range(NUM_DAYS_IN_MONTH)]
+    df = utils.build_random_sched(num_employees, shifts, rand_seed)
+
+    init_availability_table = utils.display_availability(df, NOW.month, NOW.year)
+    return (
+        init_availability_table,
+        init_availability_table,
+        True,  # disable the shedule tab when changing parameters
+        "availability-tab",  # jump back to the availability tab
+    )
+
+
+@app.long_callback(
+    Output("schedule-content", "children", allow_duplicate=True),
+    Output("schedule-tab", "disabled", allow_duplicate=True),
+    inputs=[
+        Input("run-button", "n_clicks"),
+        State("shifts-per-employee-select", "value"),
+        State("employees-per-shift-select", "value"),
+        State("checklist-input", "value"),
+        State("consecutive-shifts-select", "value"),
+        State("availability-content", "children"),
+    ],
+    running=[
+        # show cancel button and hide run button, and disable and animate results tab
+        (Output("cancel-button", "style"), {"display": "inline-block"}, {"display": "none"}),
+        (Output("run-button", "style"), {"display": "none"}, {"display": "inline-block"}),
+        # switch to schedule tab while running
+        (Output("schedule-tab", "disabled"), False, False),
+        (Output("tabs", "value"), "schedule-tab", "schedule-tab"),
+    ],
+    cancel=[Input("cancel-button", "n_clicks")],
+    prevent_initial_call=True,
 )
 def submitter(
```

Comment:
I wonder if we should come up with a consistent name for all these functions across the demos? Let me know your thoughts

Reply:
Good idea. I think it's generally fine having slightly different names between demos if it makes functionality clearer, but if they are functionally the same we should try to be consistent with naming.

Reply:
Changed to `run_optimization`, same as in MVRP.